### ISSUE TITLE
Unable to filter video search results by author/channel ID

### ISSUE THREAD

**Issue by sawyer07** [trizen/straw-viewer#34]
Today I'm migrating my scripts from youtube-viewer to straw-viewer. So far, the transition is working out quite smoothly.

In my youtube menu scripts I used to have a "more videos" option that uses youtube-dl to extract the channel ID from the most recently watched video, then launches the following in a terminal:

```youtube-viewer -C --author="$channelID" --order="date" '*' ```

I have currently rewritten this as follows:

```straw-viewer --author="$channelID" --order="upload_date" '*' ```

However, this returns no results. I am testing with [a channel ID that I verified exists according to the instance I'm connected to](https://invidious.snopyta.org/channel/UCVls1GmFKf6WlTraIb_IaJg), and tried with `--api=auto` as well. 

I also tried searching for some actual obvious search terms in place of "*" just in case that didn't work anymore, as well as typing out the literal commands directly into the command line without any variables.

In short it seems I am in general unable to filter straw-viewer search results by author using a given channel ID. As the URL provided above confirms, I am able to navigate a channel's videos in the browser. I assume their API provides this function after all?

I greatly appreciate your work and would be most grateful for any workarounds anyone might be able to offer. Thanks so much

**Comment by trizen** [comment]
Listing the latest videos from a given channel ID or channel username, seems to work correctly:

```console
straw-viewer -u UCVls1GmFKf6WlTraIb_IaJg  --debug --api=invidious.snopyta.org
```

However, the `--author` option seems to be broken in invidious itself.

From the API documentation, we have:

```
/api/v1/channels/search/:ucid
```

with parameters:

```
q: String
page: Int32
```

but https://invidious.snopyta.org/api/v1/channels/search/UCVls1GmFKf6WlTraIb_IaJg?q=ubuntu returns an empty list.

It's nothing that can be done in `straw-viewer` in this case. You may want to report this issue to invidious (if not already reported).
